include/ht_component_factory.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <string>
+#include <ht_component.h>
```

Comment:
Can forward declare Component type since only a pointer is referenced in this file.  Include this header in the file that actually needs the component interface.